extensions/wikia/FandomCreator/Hooks.php

```diff
@@ -9,11 +9,59 @@
 class Hooks {
 	const SERVICE_NAME = "content-graph-service";
 
-	public static function onNavigationApiGetData(WikiaDispatchableObject $dispatchable) {
-		self::api();
+	public static function onNavigationApiGetData( WikiaDispatchableObject $dispatchable, array $maxElementsPerLevel ) {
+		$sitemap = self::api()->getSitemap();
+		if ( $sitemap === null || !isset( $sitemap->home->children ) ) {
+			return;
+		}
+
+		$sitemapData = [];
+		foreach ( $sitemap->home->children as $i => $child ) {
+			if ( $i >= $maxElementsPerLevel[0] ) {
+				break;
+			}
+
+			$nextData = self::convertToSitemapData( $child, 1, $maxElementsPerLevel );
+			if ( $nextData !== null ) {
+				$sitemapData[] = $nextData;
+			}
+		}
+
+		$dispatchable->getResponse()->setData( [
+				'navigation' => [
+						'wikia' => [],
+						'wiki' => $sitemapData
+				]
+		] );
+	}
+
+	private static function convertToSitemapData( $entry, $currentLevel, $maxElementsPerLevel ) {
+		$numLevels = count( $maxElementsPerLevel );
+		if ( $currentLevel > $numLevels ) {
+			return null;
+		}
+
+		$data = [
+				'text' => $entry->name,
+				'href' => self::getEntityPath( $entry->id )
```

Comment:
indentation looks off here

Reply:
I ran this through the MW stylize script, so I think it's correct.

Reply:
oh weird okay